src/helpers/contactList.spec.js

```diff
@@ -0,0 +1,39 @@
+import { categorizeContacts } from './contactList'
+
+describe('Categorize contacts', () => {
+  it('should categorize contacts by indexes.byFamilyNameGivenNameEmailCozyUrl', () => {
+    const contacts = [
+      { name: 'Alex', indexes: { byFamilyNameGivenNameEmailCozyUrl: 'A' } },
+      { name: 'Alan', indexes: { byFamilyNameGivenNameEmailCozyUrl: 'A' } },
+      { name: 'Cleo', indexes: { byFamilyNameGivenNameEmailCozyUrl: 'C' } },
+      { name: 'Bernard', indexes: { byFamilyNameGivenNameEmailCozyUrl: 'B' } },
+      { name: 'Xavier', indexes: { byFamilyNameGivenNameEmailCozyUrl: 'X' } },
+      { name: 'Zorro', indexes: { byFamilyNameGivenNameEmailCozyUrl: 'Z' } },
+      { name: '', indexes: { byFamilyNameGivenNameEmailCozyUrl: '' } },
+      { name: {}, indexes: { byFamilyNameGivenNameEmailCozyUrl: {} } }
+    ]
+    const categorizedContacts = categorizeContacts(contacts, 'EMPTY')
+    expect(categorizedContacts).toEqual({
+      A: [
+        { indexes: { byFamilyNameGivenNameEmailCozyUrl: 'A' }, name: 'Alex' },
+        { indexes: { byFamilyNameGivenNameEmailCozyUrl: 'A' }, name: 'Alan' }
+      ],
+      B: [
+        { indexes: { byFamilyNameGivenNameEmailCozyUrl: 'B' }, name: 'Bernard' }
+      ],
+      C: [
+        { indexes: { byFamilyNameGivenNameEmailCozyUrl: 'C' }, name: 'Cleo' }
+      ],
+      EMPTY: [
+        { indexes: { byFamilyNameGivenNameEmailCozyUrl: '' }, name: '' },
+        { indexes: { byFamilyNameGivenNameEmailCozyUrl: {} }, name: {} }
+      ],
```

Comment:
weird to have this `empty` in the middle of the categories. Shouldn't we put in at the end?

Reply:
No I don't think so, the contactList doesn't sort anymore, but categorize only.

Reply:
So merging this PR will make the contacts list unordered? If yes, I think even for a dev release this is not wanted?

Reply:
@JF-Cozy ?

Reply:
Yes it is, it's sorted based on `_id`.

For me it's a PR among others, which are all part of a feature. As long as the last PR of the feature is not merged, there is not the expected final behavior. But I understand that it's tricky to have an unexpected behavior even if it's temporary on a Master. I will correct it and use the part of this commit on the other PR.

Reply:
If you think you need to create several PR for one feature, you can:
- Create a feature branch like `newSort`
- Creates PR on this new branch 
- Once each PR are merged in this `newSort` branch, create a PR of this branch to master. 

We don't have a development branch, we should be able to deploy what we have on master at any time

Reply:
Done :)